src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_SKCanvasElement.skia.cs

```diff
@@ -83,6 +83,79 @@ public async Task When_Waiting_For_Another_Thread2()
 		Assert.IsFalse(SUT.RenderOverrideCalledNestedly);
 	}
 
+	[TestMethod]
+	[GitHubWorkItem("https://github.com/unoplatform/uno-private/issues/1380")]
+	public async Task When_Loading_Not_Rendering_Unnecessarily()
+	{
+		var renderInvalidatedCount = 0;
+
+		var SUT = new InvalidateOnRenderSKCanvasElement
+		{
+			Height = 400,
+			Width = 400
+		};
+
+		var border = new Border
+		{
+			BorderBrush = Microsoft.UI.Colors.Green,
+			Height = 400,
+			Child = new ScrollViewer
+			{
+				VerticalAlignment = VerticalAlignment.Top,
+				Height = 400,
+				Background = Microsoft.UI.Colors.Red,
+				Content = SUT
+			}
+		};
+
+		await UITestHelper.Load(border);
+
+		SUT.XamlRoot.RenderInvalidated += () =>
+		{
+			renderInvalidatedCount++;
+		};
+
+		// Wait a short time to ensure no extra invalidations occur
+		await Task.Delay(6000);
+
+		Assert.IsTrue(renderInvalidatedCount < 4, "RenderInvalidated should not be executed indefinitely.");
+	}
+
+
+	[TestMethod]
+	[GitHubWorkItem("https://github.com/unoplatform/uno-private/issues/1380")]
+	public async Task When_Invalidate_Called_MultipleTimes_DoesNot_Crash()
+	{
+		var SUT = new InvalidateOnRenderSKCanvasElement
+		{
+			Height = 200,
+			Width = 400
+		};
+
+		var blueCanvas = new BlueFillSKCanvasElement
+		{
+			Height = 200,
+			Width = 400
+		};
+
+		var stack = new StackPanel
+		{
+			VerticalAlignment = VerticalAlignment.Stretch,
+			HorizontalAlignment = HorizontalAlignment.Stretch,
+			Height = 500,
+		};
+
+		stack.Children.Add(SUT);
```

Comment:
It would be best to clear the tree forcibly (in a try/finally) to avoid the control to stay active.